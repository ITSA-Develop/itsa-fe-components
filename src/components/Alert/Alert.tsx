import { AlertProps, Alert as AntAlert } from 'antd';

export const Alert = ({ ...rest }: AlertProps) => {
	return (
		<>
			<p>Alert!</p>
			<AntAlert {...rest} />
		</>
	);
};
