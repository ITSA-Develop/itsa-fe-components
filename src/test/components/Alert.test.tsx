import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Alert } from '../../components/Alert/Alert';

describe('Alert component', () => {
	it('renders with message', () => {
		const { container } = render(<Alert message="Este es un mensaje" />);
		expect(screen.getByText('Este es un mensaje')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('renders with description', () => {
		const { container } = render(<Alert message="Mensaje" description="Descripción detallada" />);
		expect(screen.getByText('Descripción detallada')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('renders with different types', () => {
		const types = ['success', 'info', 'warning', 'error'] as const;

		types.forEach(type => {
			const { unmount } = render(<Alert message={`Mensaje ${type}`} type={type} />);
			expect(screen.getByText(`Mensaje ${type}`)).toBeInTheDocument();
			unmount();
		});
	});
});
