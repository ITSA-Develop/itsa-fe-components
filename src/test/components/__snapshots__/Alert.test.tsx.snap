// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert component > renders with description 1`] = `
<div>
  <p>
    Alert!
  </p>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description ant-alert-no-icon css-dev-only-do-not-override-l9pxc0"
    data-show="true"
    role="alert"
  >
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Mensaje
      </div>
      <div
        class="ant-alert-description"
      >
        DescripciÃ³n detallada
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert component > renders with message 1`] = `
<div>
  <p>
    Alert!
  </p>
  <div
    class="ant-alert ant-alert-info ant-alert-no-icon css-dev-only-do-not-override-l9pxc0"
    data-show="true"
    role="alert"
  >
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Este es un mensaje
      </div>
    </div>
  </div>
</div>
`;
